src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -748,11 +748,24 @@ def _resolve_policy_id(policy, client):
     return policy_id
 
 
-def create_policy_definition(name, rules, display_name=None, description=None):
-    if os.path.exists(rules):
-        rules = get_file_json(rules)
-    else:
-        rules = shell_safe_json_parse(rules)
+def _load_file_string_or_uri(file_or_string, uri, name, required=True):
+    if file_or_string is not None and uri is not None:
+        raise CLIError('Only one of --{} or --{}-uri can be specified'.format(name, name))
+    if required:
+        if not file_or_string and not uri:
+            raise CLIError('One of --{} or --{}-uri is required'.format(name, name))
+    if file_or_string is not None:
+        if os.path.exists(file_or_string):
+            return get_file_json(file_or_string)
+        return shell_safe_json_parse(file_or_string)
+    if uri is not None:
+        with urlopen(uri) as response:
+            return json.loads(response.read())
+    return None
+
+
+def create_policy_definition(name, rules=None, rules_uri=None, display_name=None, description=None):
+    rules = _load_file_string_or_uri(rules, rules_uri, 'rules')
```

Comment:
If rules can accept a JSON string of local path, then you should just add a third overload for URI instead of creating another parameter. It should be easy to sniff the difference between a local file and URI.